### PR TITLE
feat (edit-cra): adding the billing form

### PR DESCRIPTION
**Pull request by mrclemds**
add the fields that make up the billing form
add the corresponding scss

### CODE REVIEW

File: src/app/cras/edit-cra/invoice-form/invoice-form.component.html [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                      name="invoiceBankDetailsDomiciliationInput"
+                      formControlName="invoiceBankDetailsDomiciliationInput"
+                      >
+                  </div>
+              </div>
+          </div>
+        </div>
+    </div>
+  </div>
+</div>
```
Comment (pfilippi) [inline src/app/cras/edit-cra/invoice-form/invoice-form.component.html]: Templates are getting big; Shouldn't we use directives to make them more readable?

File: src/app/shared/invoice/invoice-data.service.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return this.invoice;
+  }
+
+  public setInvoice(invoice: Invoice, update = false): void {
+    this.invoice = invoice;
+    if (update) {
+      this.dataChanged.next(this.getInvoice());
+    }
+  }
+}
```
Comment (pfilippi) [inline src/app/shared/invoice/invoice-data.service.ts]: I don't understand what this service is for

File: src/app/@types/invoice.ts [hunk 3]
```diff
@@ -1,16 +1,16 @@
-import { BillPart } from '../shared/billPart.model';
+import { InvoicePart } from '../shared/invoice-part.model';
```
Comment (pfilippi) [inline src/app/@types/invoice.ts]: "InvoicePart" is not the most obvious name. "Company" would ne more straightforward

File: src/app/@types/invoice.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
-export type Bill = {
+export type Invoice = {
   number: number;
   date: string;
   clientRef: string;
   dailyRate: number;
-  consultant: BillPart;
-  client: BillPart;
+  consultant: InvoicePart;
```
Comment (pfilippi) [inline src/app/@types/invoice.ts]: This is no consultant. "Provider"

File: src/app/cras/edit-cra/invoice-form/invoice-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        }
+
+        this.invoiceDataService.setInvoice(this.invoice);
+      });
+    });
+
+    return this.formBuilder.group(controls);
+  }
+
+  intAutoFormUpdater(): void {
```
Comment (pfilippi) [inline src/app/cras/edit-cra/invoice-form/invoice-form.component.ts]: I dont understand this method

File: src/app/shared/invoice/invoice-data.service.ts (hunk 2 above)
Comment (mrclemds) [reply src/app/shared/invoice/invoice-data.service.ts]: This service simulates the two-way binding, between two component. It is useful when you change the URL data using the routing feature.

File: src/app/cras/edit-cra/invoice-form/invoice-form.component.ts (hunk 5 above)
Comment (mrclemds) [reply src/app/cras/edit-cra/invoice-form/invoice-form.component.ts]: This method subscribes to the observable that changes each time another component changes the invoice. Then update the form.

Comment (pfilippi) [reply src/app/cras/edit-cra/invoice-form/invoice-form.component.ts]: Having to manually update things when data changes seems wrong.

**Pull request merged**